shepherd/Timer.py

```diff
@@ -1,8 +1,54 @@
+import time
+import threading
+import collections
+import heapq
+import LCM
+import utils
+
 class Timer:
     """
     This class should spawn another thread that will keep track of a target time
     and compare it to the current system time in order to see how much time is left
     """
+    class busyThread(threading.Thread):
+    '''
+    Subclass that is the actual thread that will be running.
+    There will only be one for the entire class.
+    '''
+    def __init__(self, queue):
+        super(busyThread, self).__init__()
+        self.queue = queue
```

Comment:
Actually, you probably can simplify all this synchronization logic by using the python built in `Queue.PriorityQueue`